src/gen_alpha_poc/generalized_alpha_time_integrator.cpp

```diff
@@ -226,38 +216,43 @@ std::tuple<State, Kokkos::View<double*>> GeneralizedAlphaTimeIntegrator::AlphaSt
                 Kokkos::parallel_for(
                     n_constraints,
                     KOKKOS_LAMBDA(const size_t i) {
-                        // Take negative of the solution increments to update Lagrange multipliers
+                        // Take negative of the solution increments to update Lagrange
+                        // multipliers
                         delta_lagrange_mults(i) = -soln_increments(i + delta_gen_coords.size());
                         lagrange_mults_next(i) += delta_lagrange_mults(i);
                     }
                 );
             }
         }
 
-        // Update the velocity, acceleration, and constraints based on the increments
-        Kokkos::parallel_for(
-            size,
-            KOKKOS_LAMBDA(const size_t i) {
-                delta_gen_coords(i) += delta_x(i) / h;
-                velocity(i) += kGammaPrime * delta_x(i);
-                acceleration(i) += kBetaPrime * delta_x(i);
-            }
-        );
+        for (size_t node = 0; node < n_nodes; node++) {
+            // Update the velocity, acceleration, and constraints based on the increments
+            Kokkos::parallel_for(
+                size,
+                KOKKOS_LAMBDA(const size_t i) {
+                    delta_gen_coords(node, i) += delta_x(i) / h;
+                    velocity_next(node, i) += kGammaPrime * delta_x(i);
+                    acceleration_next(node, i) += kBetaPrime * delta_x(i);
+                }
+            );
+
+            // Update algorithmic acceleration once Newton-Raphson iterations have ended
+            Kokkos::parallel_for(
+                size,
+                KOKKOS_LAMBDA(const size_t i) {
+                    algo_acceleration_next(node, i) +=
+                        (1. - kAlphaFLocal) / (1. - kAlphaMLocal) * acceleration_next(node, i);
+                }
+            );
+        }
```

Comment:
Similar to Derek's above comment, we can probably optimize this by using an mdrange and atomic access traits.  I am leaving this comment more as a reminder to future us rather than as an immediate TODO

Reply:
Sounds good, I'll leave this comment unresolved as well and revisit this PR to create an issue later.